Add instructions for adding new protos

BenHenning opened the pull request:
This introduces instructions for how to add new protos to the model library.

BenHenning committed:
Add instructions for adding new protos

aggarwalpulkit596 commented on model/BUILD.bazel:
```diff
@@ -7,6 +7,13 @@ The java_lite_proto_library() rule takes in a proto_library target and generates
+# bottom of this file so that other parts of the app get access to it. If protos import other
+# protos, they need to use format_import_proto_library (again, see examples below for how to do
+# this).
+
```
maybe an explicit example could be added to make it a bit simpler to understand otherwise looks good to me

BenHenning replied:
I thought about that, but I thought that the existing libraries were simple enough to follow as an example. Do you think an additional example is really needed @aggarwalpulkit596?

aggarwalpulkit596 replied:
I think for newcomers bazel would be something very different and they won't be familiar with it in the first place and that's the reason I was thinking to make a simple sample to help new contributors on board easily. We could make a sample repo as well for this and point that repo, if that sounds good to you, @anandwana001 @rt4914 your thoughts?

BenHenning replied:
Ah, in that case I agree that we should have more Bazel onboarding instructions as we move farther along (@miaboloix as well for visibility & in case you have any thoughts). That being said, do you think anything additional is needed here for protos? It seemed straightforward since the other libraries below are quite simple (and the example would more or less be exactly the same thing), but if you and/or others think an explicit example is still helpful then I'm happy to add one.

anandwana001 replied:
I think an explicit example will be helpful.

BenHenning replied:
Added an example. @anandwana001 PTAL.

anandwana001 replied:
Looks helpful to me, Thanks.

aggarwalpulkit596 replied:
Looks good to me too, Thanks.

miaboloix reviewed (approved):
LGTM!

BenHenning commented:
Thanks for the reviews! @aggarwalpulkit596 sending this back your way per your comment. Also, please remember to reassign to me if you're done with your review pass (otherwise I can't follow up :)).

rt4914 reviewed (approved):
LGTM, thanks.

anandwana001 reviewed (approved):
LGTM, Thanks

BenHenning committed:
Update BUILD.bazel

Add sample proto to further explain how to add properly add a new proto to the codebase.

rt4914 commented:
@BenHenning @anandwana001 Should we merge this PR?

BenHenning commented:
@rt4914 did you miss my comment thread above asking @anandwana001 to double check the updates? Please make sure to double check comment threads before merging per https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#tips-for-getting-your-pr-merged-after-submission.

BenHenning commented:
Also, @anandwana001 please take a look and, if there are any changes necessary, please let me know so that I can address them.

rt4914 commented:
> @rt4914 did you miss my comment thread above asking @anandwana001 to double check the updates? Please make sure to double check comment threads before merging per https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#tips-for-getting-your-pr-merged-after-submission.

@BenHenning I remember seeing the above thread but don't exactly remember why I still choose to merge it. Apologies, I will keep this in mind. Thanks

rt4914 commented:
@BenHenning Not reverting this as of now and will wait @aggarwalpulkit596 and @anandwana001 to address the comment and decide to resolve or not.

anandwana001 commented:
> @rt4914 did you miss my comment thread above asking @anandwana001 to double check the updates? Please make sure to double check comment threads before merging per https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#tips-for-getting-your-pr-merged-after-submission.

ah, Rajat and Me had a quick discussion over this yesterday before the team meeting. Delayed to left comment as it got late yesterday. The example looks good to me.
